pkg/core/consensus/selection/step.go

```diff
@@ -142,17 +142,19 @@ func (p *Phase) Run(parentCtx context.Context, queue *consensus.Queue, evChan ch
 
 	for _, ev := range queue.GetEvents(r.Round, step) {
 		if ev.Category() == topics.Score {
-			p.collectScore(ctx, ev.Payload().(message.Score))
+			p.collectScore(ctx, ev.Payload().(message.Score), ev.Header(), true)
 		}
 	}
 
 	for {
 		select {
 		case internalScoreResult := <-internalScoreChan:
-			p.collectScore(ctx, internalScoreResult.Payload().(message.Score))
+			// disable topics.Score network republish here as it's been already published in generateCandidate.
+			const republish = false
+			p.collectScore(ctx, internalScoreResult.Payload().(message.Score), internalScoreResult.Header(), republish)
```

Comment:
As discussed before on the standup, I think it would be cleaner to remove the republishing in the candidate generator, and then avoid the boolean flag since we can always republish in `collectScore`. As checked with Toghrul, it's legal in the protocol to avoid republishing one's own `Score` message in case a better one was already processed.

Reply:
Indeed. I should probably remove republish flag from this PR and handle the "optimization" with a separate issue/PR. This is because a such protocol-related change should pass 10K-round test to ensure no regressions occur. I still have a hunch that we had similar idea implemented time ago and that had negative impact on the consensus average time. Need to test this more.

Reply:
@julesdesmit Feel free to review description of #1048